workflow/executor.py

```diff
@@ -286,6 +287,26 @@ def execute(self, logger: logging.Logger, env_config: EnvConfig, rai_config: Rai
         for src in missed_resources:
             self._load_source(logger, env_config, rai_config, src)
 
+        self.await_pending(env_config, logger, missed_resources)
+
+    def await_pending(self, env_config, logger, missed_resources):
+        loop = asyncio.get_event_loop()
+        if loop.is_running():
+            raise Exception('Waiting for resource would interrupt unexpected event loop - aborting to avoid confusion.')
```

Comment:
An alternative would be to make calling functions async all the way up the call stack. This seemed like an architectural decision you should make, so I chose the less intrusive approach. The downside is - if the exception above were removed - the async work I added could be interleaved with other async work and interrupt this other work.